src/Analyzer.BicepProcessor/BicepTemplateProcessor.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Bicep.Core.Analyzers.Linter;
+using Bicep.Core.Configuration;
+using Bicep.Core.Emit;
+using Bicep.Core.Features;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Registry;
+using Bicep.Core.Registry.Auth;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.TypeSystem.Az;
+using Bicep.Core.Workspaces;
+
+namespace Microsoft.Azure.Templates.Analyzer.BicepProcessor
+{
+    /// <summary>
+    /// Contains functionality to process Bicep templates.
+    /// </summary>
+    public class BicepTemplateProcessor
+    {
+        /// <summary>
+        /// Converts Bicep template into JSON template and returns it as a string
+        /// </summary>
+        /// <param name="bicepPath">The Bicep Template file path.</param>
+        /// <returns>The compiled template as a <c>JSON</c> string and its source map.</returns>
+        static public (string, SourceMap) ConvertBicepToJson(string bicepPath)
+        {
+            using var stringWriter = new StringWriter();
+
+            Environment.SetEnvironmentVariable("BICEP_SOURCEMAPPING_ENABLED", "true");
+
+            var fileResolver = new FileResolver();
+            var featureProvider = new FeatureProvider();
+            
+            var moduleDispatcher = new ModuleDispatcher(
+                new DefaultModuleRegistryProvider(
```

Comment:
I don't know how important the module registry is for the analyzer, but it might be worth verifying what the behavior is like if you attempt to access a .bicep file from a module registry or a template spec - e.g.
```bicep
module gridVnet 'br:Carml.azurecr.io/bicep/modules/microsoft.network.virtualnetworks:v0.4.0' = {
  name: 'foo'
}
```

Reply:
Yeah, as expected I suppose causes issues when building JSON:
`The module with reference &quot;br:carml.azurecr.io/bicep/modules/microsoft.network.virtualnetworks:v0.4.0&quot; has not been restored.`

Reply:
I don't see any other implementations for IModuleRegistryProvider, and I am also not sure how this would work with the source map either. Do you have a little more context here or thoughts about how we should be handling these sorts of bicep files?

Reply:
I think best you can do is handle the case nicely where authentication fails. I believe if you were to log in with az CLI or az powershell before building the Bicep file, and you have access to the registry, module restore should work, but there will definitely be cases where the registry is private and auth isn't possible.

Reply:
If you do want to be able to pull down modules optimistically, you should be able to use:
https://github.com/Azure/bicep/blob/d46c7d2ee2c78afc0dad767fa21f78b352a7d4d1/src/Bicep.Cli/Services/CompilationService.cs#L58-L64

Reply:
OK added the restore functionality, though I wasn't able to auth to this registry still. Will look into soon as modules are not going to be initially supported until the nested template work in progress is closer to complete.